nncf/scopes.py

```diff
@@ -130,78 +130,142 @@ def convert_ignored_scope_to_list(ignored_scope: Optional[IgnoredScope]) -> List
+            matches["types"].add(node.node_name)
+            types.add(node.node_type)
+
+        for i, subgraph in enumerate(ignored_scope.subgraphs):
+            names_from_subgraph = get_ignored_node_names_from_subgraph(graph, subgraph)
+            if names_from_subgraph:
+                matches["subgraphs"].update(names_from_subgraph)
+                subgraphs_numbers.add(i)
+
+    matched_ignored_scope = IgnoredScope(
+        names=list(names),
+        patterns=list(patterns),
+        types=list(types),
+        subgraphs=[subgraph for i, subgraph in enumerate(ignored_scope.subgraphs) if i in subgraphs_numbers],
+        validate=ignored_scope.validate,
+    )
+    return matched_ignored_scope, matches
+
+
+def get_unmatched_ignored_scope(matched_ignored_scope: IgnoredScope, ignored_scope: IgnoredScope) -> IgnoredScope:
+    """
+    Returns unmatched ignored scope rules from full ignored scope and matched ignored scope.
+
+    :param matched_ignored_scope: Matched ingored scope.
+    :param ignored_scope: Full ignored scope.
+    :return: Unmatched ignored scope.
+    """
+    assert matched_ignored_scope.validate == ignored_scope.validate
+    return IgnoredScope(
+        names=[name for name in ignored_scope.names if name not in matched_ignored_scope.names],
+        patterns=[pattern for pattern in ignored_scope.patterns if pattern not in matched_ignored_scope.patterns],
+        types=[type for type in ignored_scope.types if type not in matched_ignored_scope.types],
+        subgraphs=[subgraph for subgraph in ignored_scope.subgraphs if subgraph not in matched_ignored_scope.subgraphs],
+        validate=matched_ignored_scope.validate,
+    )
+
+
+def info_matched_ignored_scope(matches: Dict[str, Set[str]]) -> None:
+    """
+    Log matches.
+
+    :param matches: Matches.
+    """
+    for rule_type, rules in matches.items():
+        if rules:
+            nncf_logger.info(f"{len(rules)} ignored nodes were found by {rule_type} in the NNCFGraph")
+
+
+def error_unmatched_ignored_scope(unmatched_ignored_scope: IgnoredScope) -> str:
+    """
+    Returns an error message for unmatched ignored scope.
+
+    :param unmatched_ignored_scope: Unmatched ignored scope.
+    :return str: Error message.
+    """
+    err_msg = ""
+    if unmatched_ignored_scope.names:
+        err_msg += f"Ignored nodes with name {unmatched_ignored_scope.names} were not found in the NNCFGraph. "
+    if unmatched_ignored_scope.patterns:
+        err_msg += f"No matches for ignored patterns {unmatched_ignored_scope.patterns} in the NNCFGraph. "
+    if unmatched_ignored_scope.types:
+        err_msg += f"Nodes with ignored types {unmatched_ignored_scope.types} were not found in the NNCFGraph. "
+    for subgraph in unmatched_ignored_scope.subgraphs:
+        err_msg += (
+            f"Ignored subgraph with input names {subgraph.inputs} and output names {subgraph.outputs} "
+            "was not found in the NNCFGraph. "
+        )
```

Comment:
The base of the message looks the same. For example:
`f"Ignored node that matches {ignored_type = "names, types, patterns, subgraphs"} were not found in the NNCFGraph. \n"`
Can be used for each case.